source/common/network/utility.cc

```diff
@@ -66,13 +66,30 @@ std::string hostFromUrl(const std::string& url, absl::string_view scheme,
     throw EnvoyException(fmt::format("expected {} scheme, got: {}", scheme_name, url));
   }
 
-  const size_t colon_index = url.find(':', scheme.size());
+  const size_t bracket_begin_index = url.find('[', scheme.size());
 
-  if (colon_index == std::string::npos) {
-    throw EnvoyException(absl::StrCat("malformed url: ", url));
-  }
+  if (bracket_begin_index == std::string::npos) {
+    const size_t colon_index = url.find(':', scheme.size());
```

Comment:
Could you include some comments here that explains the logic a bit? It's a bit hard to follow just from the code

Reply:
got it, let me add the comment, thanks for the review!